build.assets/genproto.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# Builds, formats, lints and generates protos for teleport and teleport/api.
+set -eu
+
+echoed() {
+  echo "$*" >&2
+  "$@"
+}
+
+main() {
+  cd "$(dirname "$0")"  # ./build-assets/
+  cd ../                # teleport root
+
+  buf build
+  buf lint
+  buf format -w
+
+  # Generated protos are written to
+  # <teleport-root>/github.com/gravitational/teleport/..., so we copy them to
+  # the correct relative path.
+  buf generate
+  find github.com -name '*.pb.go' | while read -r f; do
+    mv "$f" "${f#github.com/gravitational/teleport/}"
```

Comment:
Is this step truly necessary or is there a way to have `but` just put stuff in the right place?

Reply:
I had no problems making protoc-gen-go do my bidding, but gogofast seems to ignore my attempts to make it behave. I'll see if I can come up with any other ideas.

(TBH I even considered doing a temporary `ln -s teleport github.com/gravitational/teleport`, it's dirty but it's so easy...)

Reply:
Circling back to this, because legacy protos don't to directory->package correspondence correctly, no output mode quite nails the structure we need. Writing the files to their final go_package paths and then moving seems to be easiest way to support it.